### ISSUE TITLE
FGP Viewer Usability Testing Results

### ISSUE THREAD

**Issue by mweech** [fgpv-vpgf/fgpv-vpgf#1401]
The following feedback has been received following usability tests conducted by FGP Web Presence team. This is feedback and will be analyzed for possible CR's to improve usability of the viewer.

All items below will be either:

- have an issue opened, added to this epic, and noted on the list item with a checkbox
- or an explanation will be given inline if they will not be directly addressed.

### V1. Arrival disorientation (Type:Functionality)

When testers arrived at the map from the catalogue or the map cart they expected to see a list of the layers that theyhad chosen.  Because the legend startsout closed they did not see their layers and were therefore disoriented, wonderingwhether they were in the “right place”.

#### Comments:

- Feature is currently in the backlog under issue #1338.  
- To be scheduled for implementation under v1.5

### V2. Confusion caused by the menuicon placement (Type: Design)

Testers repeatedly clicked on the menu iconwhen they wanted to open the legend and were frustrated to continually make thesame mistake.  Testers commented that thelegend icon should be moved to the left hand side.

#### Comments:

- The intent of this menu button is governed by the Material design guidelines we are following and as a result there will be no direct change.  This is logged as a known issue #651.
- We will add ability to open/close the layer selector from the left-sided menu as well so there are more than one path for the user to reach the list of layers #1424 
- Also, under #1338 we are adding functionality to have the legend open by default that should mitigate some initial confusion

### V3. Legend confusion (Type: Labelling)

Testers were confused by the “Legend” tooltip and “Legend” title and felt that a more descriptive word might be“Layers” or in French “Couches”.

#### Comments: 

- This panel was initially called "Layers" and it was renamed to "Legend" because of feedback that some users were confused by what "Layers" mean.
  -- In the structured legend, it’s not a list of layers (no one-to-one relationship)
  -- If all users prefer “Layers”, we will change it
  -- Other options: "Table of content"
- Will revert back to "Layers" as title #1426 

### V4. Unable to find and open the symbology (Type: Design)

Inside the legend most testers were notaware that the symbology icons (including animations) were clickable.  After assistance some testers were able toopen it and asked that an expandable arrow be used to indicate that this partis expandable.

#### Comments:

- Will add menu option (show legend) to the layers drop down menu (this will add another way of opening the symbology stack) #1425 
- Fan out icons when hovering over a legend record?

### V5. Symbology confusion (Type: Labelling)

Once testers clicked open the symbology(most required assistance to get there) they felt a more intuitive term forthat function’s tooltip would be “Show legend” or in French “Afficher lalégende”

#### Comments:

-         Change tooltip to “Legend details” or “Expand legend" #1434 

### V6. Unable to find import function(Type: Design)

Testers were unable to find the import function without assistance, mostly because it required testers to open the legend first.  Some testers wished that the import function was out of the legend into a “top menu bar”.

#### Comments:

- Will add the import button to the side menu (call it “Add layer” instead of “import”; the tooltip on the button should say the same) #1425
- In cases when the legend is open by default, the import button is readily available
- Add empty state #143 

### V7. Plus icon confusion (Type: Labelling)

Once the testers opened the legend they needed to interpret the plus icon to mean import; some testers asked that the icon be replaced by the text “Import”.

#### Comments:

- There is not sufficient room for text to be added.  This function will also be available from the side menu with a text label (V6).  #1425 
- Update tooltip to "Add layer" #1434 

### V8. Import final “Continue” buttonconfusion (Type: Functionality)

After pressing the final “Continue” toimport a layer, there is no indicator that the application is working, so the testers pressed the “Continue” button several times and wondered if it wasworking.

#### Comments:

- Progress indicator is being blocked by JS crunching numbers when importing a layer. UI thread is being blocked and all animations (expect for CSS transform animation are stalled).
- Options:
-- Add a notification explaining the app might feel unresponsive while data is processed (quick!)
-- Custom css animation which is not blocked
-- Offload processing to a web-worker (IE might support web-workers in a special way)
- #1428 

### V9. Unable to open table (Type: Labelling)

Testers were unable to open the table.  After stumbling upon the table (by randomlyclicking) they were unable to open it a second time. In some cases testers leftthe test without opening the table on the first try (only finding it each time byclicking around randomly until the table opened).

#### Comments:

- Add option to the layer dropdown menu to open datatable
- Bring back the layer flag icon indicating the layer has data 
- #1429 

### V10.  Unable to find basemap switcher (Type: Labelling)

Testers were unable to find the basemapswitching function and commented that the icon didn’t mean anything to them.  Testers also asked that the tooltip be clearer such as “Basemap options” or “Change the basemap” or “Switch the basemap”.

#### Comments:

- Add basemap toggle to the sidemenu #1424 
- Put basemap switcher button on the app bar with the import button or as an item in the layers list itself
     -  in this case, basemap selector will only be available when the legend is open. (Johann noted it works like this in GeoCanViz and no one complained so far)

### V11.  Unable to close basemap switcher (Type:Functionality)

Once testers found the basemap switcher andused it successfully they were unable to close the basemap switcher.  Most testers expected the basemap switcher toslide closed once they moved their mouse off of it.  An alternative was to have a title bar at thetop such as “Basemap Options” with an “X” to close the window.

#### Comments:

- Fixed by #1177

### V12. Point information trigger confusion (Type: Labelling)

Testers were startled when the pointinformation box appeared and uncertain what had triggered it.  Testers indicated that a cursor change orhover tooltip would be a good signal that features on the map are clickable formore information.

#### Comments:

- Wont fix (https://github.com/fgpv-vpgf/fgpv-vpgf/issues/424#issuecomment-221270498)
- Tooltips are non-trivial to implement and will have a significant impact on user experience for a catalogue viewer (will be implemented for thematic feature layer based maps only)
- Behaviour will become more familiar after initial learning curve.

### V13. Unable to close point information box (Type: Design)

Once the point information box appeared testerswere unable to close it.  They would flipback and forth between the legend and the point information.  They also tried clicking on the map but thatdidn’t close it either.  Most testers did not see the “X” to close the point information box without assistance.

#### Comments:

-         Exploring idea of moving panel titles inside the panel (having a close button on the same line as title might help) #1431           

### V14. Menu confusion (Type: Design)

Testers kept opening the menu accidentally while looking for the legend.  Some expressed that they felt that the far left hand menu was too prominently positioned for the amount of functionality it contains.

#### Comments:

- Options here will grow over time and increase the significance.  We are already planning to add the Add layer dialogue, and additional make open/close of layer list available here as well

### V15.  Help confusion (Type: Design)

Some testers were unable to find the helpmenu and some testers expressed that it should be located more prominently.  The general expectation wasthat the help would be found in the top right or the bottom right of the viewer

#### Comments:

- Will add a help button to the nav control cluster #1432 

### V16. Thin red vertical bar in legend (Type: Labelling)

At some points in the usability tests, a thin red vertical bar appears beside someof the layers in the legend section and the testers would ask why (maybe it hasto do with the point information functions?). Testers tried to scroll it, and tried to click on it and still did notknow what it was for.

#### Comments:

- Intermittent bug that needs fixing and may need to be fixed in relation to Thematic Map refactoring #1433 

### V17. Point information confusion (Type: Labelling)

Testers were confused why it was called “Point Information” when it can also contain information about polygon and line features.

#### Comments:

- Will change title to "Feature Details" #1434 

### V18.  Table “jumping” (Type: Functionality)

When using the tables, testers found that whensorting a column of information the column may “jump” sideways.   So the testers found that if you want to flip back and forth between ascending and descending sort order you have to keep moving your mouse or you will sort on a different column.  Also the testers found the same table “jumping” happens when scrolling through a table.

#### Comments:

- This is a technical limitation of the Datatable plugin as it is not fully aware of the max width of the data until it is encountered, unless all data is preloaded and processed for these widths (expensive processing).
- Wont' fix.  There is sufficient effort and performance tradeoffs to address this vs. the business value of situations where a user would repeatedly change the sort order by clicking multiple times in a row.  In typical usage, the user might change the sort 1-2 times before pausing and performing other actions or looking at the data.

- Set fixed column width based on service configuration (https://github.com/fgpv-vpgf/fgpv-vpgf/issues/1482)

### V19.  Your location (Type: Functionality)

No tester was able to make the “Your location” button do something.

#### Comments:

- Fixed under #954.

### V20.  Full screen state change (Type:  Labelling)

After pressing the full screen button, when a tester wanted to return to the regular page “non-full screen” state the button is still labelled full screen.

#### Comments:

- There is an existing issue for this: #559 
- Will add a checkmark icon to the full-screen option when it's on #1424 

### V21.  Disappearing get link and map cart (Type:Functionality)

After pressing the full screen button, testers found that the get link and map cart buttons are no longer visible or usable.

#### Comments:

- Viewer is no longer in control of host page.  If viewer is 100% responsible for generation of state used for bookmarks and back to cart, there will be cases where languages will be improperly captured.  
-- Will add functionality to provide this to the core and move it into an external plugin when plugin api is implemented (will allow for greater control of how the cart and bookmark functions work at that time)
-- Will document known cases where the link would not work as expected when the viewer is generating state instead of the host page (add to the wiki)
#1438 

### V22.  Missing tooltips (Type: Labelling)

Some buttons do not have a tooltip, such as the “…” and the vertical “…”.  Some testers navigated almost entirely using tooltips and commented every time a tooltip was missing, “no hover information, wonder what this will do”.

#### Comments:

- Material design guidelines govern that menu buttons should not have tooltips.  There are aria tags for accessibility.  Won't fix.

### V23.  Get link confusion (Type: Labelling)

Testers were looking for text such as “Share link” with a recognizable icon such as (box with arrow icon) or (less than symbol with dots icon) instead of the “Get Link” button.

#### Comments:

- Will change label to "Share" without "link", as this is commonly used.  Will utilize the material design icon for share (< with 3 dots) #1434 

### V24. Place name search (Type:Functionality)

Testers wanted to be able to search for alocation by name and zoom then to it on the map.

#### Comments:

- Being implemented under #186.  Slated for release in v1.6.
-- Functionality referred to here is done.

### V25. Highlight record and feature (Type: Functionality)

Testers wanted to be able to select a recordin a table that would then be highlighted in the table as well as highlighting that feature on the map.

#### Comments:

- This is already in the backlog for enhanced datagrid #884 

### V26. Toggle layer visibility all on / all off (Type: Functionality)

Testers wanted to be able to turn the visibility on and off of all layers in the legend with a single toggle button.

#### Comments:

- Putting a toggle on the app bar when the legend is open only (along with the import layer button)
- Considering a separate line at the top of the legend with the toggle button and as a possible place to hold future functionality (mockups to come)
- #1440

### V27.  Filter features using symbology (Type:Functionality)

Testers wanted to be able to use thelegend/symbology items to filter the map as well as the table.  For example, in the dataset, ????

#### Comments:

- Assume this is to do with filtering based on individual layers symbology sets.  
- Worth logging for addition to the backlog and a possible future enhancement #1442

### V28.  Print map (Type: Functionality)

Testers wanted to be able to print their mapincluding a title, scale bar, north arrow, legend, etc.

#### Comments:

- Will be available in 1.4 release #176 

### V29. Help navigation (Type: Wayfinding)

Testers wanted to be able to keyword search the help or if not possible then to navigate by sections using a table of contents with hyperlinks.

#### Comments:

- Will implement collapsible sections for now #1443
- Create issue to add search in the help.  Eventual implementation of contextual help might make this less of an issue. #1444
